services/web/server/src/simcore_service_webserver/functions/_controller/_functions_rest.py

```diff
@@ -4,27 +4,48 @@
     FunctionToRegister,
     RegisteredFunction,
     RegisteredFunctionGet,
+    RegisteredFunctionUpdate,
+    RegisteredProjectFunctionGet,
 )
 from models_library.api_schemas_webserver.users import MyFunctionPermissionsGet
+from models_library.functions import FunctionClass, RegisteredProjectFunction
+from models_library.rest_pagination import ItemT, Page
+from models_library.rest_pagination_utils import paginate_data
 from pydantic import TypeAdapter
 from servicelib.aiohttp import status
 from servicelib.aiohttp.requests_validation import (
     handle_validation_as_http_error,
     parse_request_path_parameters_as,
+    parse_request_query_parameters_as,
 )
+from servicelib.mimetype_constants import MIMETYPE_APPLICATION_JSON
+from servicelib.rest_constants import RESPONSE_MODEL_POLICY
 
 from ..._meta import API_VTAG as VTAG
 from ...login.decorators import login_required
 from ...models import AuthenticatedRequestContext
+from ...projects import _projects_service
+from ...projects.models import ProjectDBGet
 from ...security.decorators import permission_required
 from ...utils_aiohttp import envelope_json_response
 from .. import _functions_service
 from ._functions_rest_exceptions import handle_rest_requests_exceptions
-from ._functions_rest_schemas import FunctionPathParams
+from ._functions_rest_schemas import (
+    FunctionGetQueryParams,
+    FunctionPathParams,
+    FunctionsListQueryParams,
+)
 
 routes = web.RouteTableDef()
 
 
+def _create_json_response_from_page(page: Page[ItemT]) -> web.Response:
```

Comment:
this looks like something that could be re-used generally.
Is there nothing like so already somewhere @pcrespov?

Reply:
yes, check `utils_aiohttp`
I am trying to put it all in there ... please do a search and if you find others, use the same. 
Thx

Reply:
Done, thanks